### PR TITLE
feat(admin-view-class): view class page

### PR DESCRIPTION
**Pull request by MatoPlus**
### Notion ticket link

[View class participants](https://www.notion.so/uwblueprintexecs/View-class-participants-1e9bb234b1744ef8b14685986d6cd59d)

### Implementation description

https://user-images.githubusercontent.com/44826218/140614423-278868bf-dc43-4196-82c4-4652d0f2b65c.mp4

- Used breadcrumb for trail at the top
- Kinda stubby class card drop down option for now (no modal options, @brandonwong368 should have an implementation to integrate)
- Two tabs, student/volunteer
- Use new library, `react-table` for sorted table and global filter
- use new library `react-csv` for csv export
- private get class registrants for admins and teachers only (TODO)

### Steps to test

1. Test flow for class with volunteers
2. Test flow for class with students
3. Global filter filters any column
4. Export csv works and the output makes sense

### What should reviewers focus on?

- Dev: Code logic
- Dev: Any induced bugs?
- Design: Look and feel

### Checklist

-   [X] My PR name is descriptive and in imperative tense
-   [X] I have run the linter
-   [X] I have requested a review from the PL, and/or other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

**Comment by MatoPlus** [comment]
Ready to review, just missing some small stuff
- link for table rows
- better names for csv export
- rbac for getRegistrant

**Review by bekahma — COMMENTED** [review]
couple of small comments/questions:
1. will the table have a sort functionality?
2. since there aren't so many rows, @vedantdsn and i wanted to make the table infinitely scrollable so it's not only 3-4 children on each page - what do you think about this?
3. can we center the age/grade field? or left align is ok too for consistency
4. from our last convo, we talked about having a tooltip beside the name for special notes for the children. wanted to confirm if that's still in the scope and i can draft something up

very cool exporting to .csv feature though 😄

**Comment by MatoPlus** [comment]
> 1. have a sort functionality?
> 2. since there aren't so many rows, @vedantdsn and i wanted to make the table infinitely scrollable so it's not only 3-4 children on each page - what do you think about this?
> 3. can we center the age/grade field? or left align is ok too for consistency
> 4. from our last convo, we talked about having a tooltip beside the name for special notes for the children. wanted to confirm if that's still in the scope and i can draft something up
> 
> very cool exporting to .csv feature though 😄

@bekahma 
1. yes it's in there, I'll update the video to show that lol. Or you can feel free to play around on the page at https://social-diversity-for-children-pr-124.up.railway.app/admin/class/10000 in the meanwhile. (You'll need to login using your admin account)
2. sounds good. (TODO Item for me)
3. Yea actually, chakra ui says that right align for numerics are sensible defaults. I think it's fine to follow their conventions to make the code a bit cleaner. https://chakra-ui.com/docs/data-display/table
4. well this page is only for students and volunteers right? We don't have a parent tab. Lmk if we want something in this page though

All great questions 👍

**Comment by MatoPlus** [comment]
> since there aren't so many rows, @vedantdsn and i wanted to make the table infinitely scrollable so it's not only 3-4 children on each page - what do you think about this?

@bekahma @vedantdsn 
Can we make a compromise and say like 20 rows per page lol. Since the pagination thing already in there :P

**Comment by MatoPlus** [comment]
Update: 

- Added better support for columns for both student and volunteer tables.

https://user-images.githubusercontent.com/44826218/140852045-8dc33e48-9d25-4d67-abe1-c9399789de91.mp4

File: pages/admin/class/[id].tsx [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+                            {classCard.name}
+                        </BreadcrumbLink>
+                    </BreadcrumbItem>
+                </Breadcrumb>
+                <ClassViewInfoCard cardInfo={classCard} />
+                <Tabs w="100%">
+                    <TabList>
+                        <Tab>
+                            {classCard.spaceTaken} Student
+                            {classCard.spaceTaken != 1 ? "s" : ""}
```
Comment (MatoPlus) [inline pages/admin/class/[id].tsx]:
```suggestion
                            {classCard.spaceTaken !== 1 ? "s" : ""}
```

File: pages/api/class/archive/[classId].ts [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+export default async function handle(
+    req: NextApiRequest,
+    res: NextApiResponse,
+): Promise<void> {
+    // Obtain class id
+    const { classId } = req.query;
+    const { isArchive }: { isArchive: boolean } = req.body;
+    const session = await getSession({ req });
+
+    // If there is no session or the user is not a internal user, not authorized
```
Comment (MatoPlus) [inline pages/api/class/archive/[classId].ts]:
```suggestion
    // If there is no session or the user is not an admin user, not authorized
```

File: pages/api/class/archive/[classId].ts [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    //parse query parameters from string to number and validate that id is a number
+    const id = parseInt(classId as string, 10);
+    if (isNaN(id)) {
+        return ResponseUtil.returnBadRequest(
+            res,
+            "classId should be passed in as numbers",
+        );
+    }
+
+    if (req.method == "PUT") {
```
Comment (MatoPlus) [inline pages/api/class/archive/[classId].ts]: use switch statements instead.

File: pages/api/class/registrant/[classId].ts [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+    //parse query parameters from string to number and validate that id is a number
+    const id = parseInt(classId as string, 10);
+    if (isNaN(id)) {
+        return ResponseUtil.returnBadRequest(
+            res,
+            "classId should be passed in as numbers",
+        );
+    }
+
+    if (req.method == "GET") {
```
Comment (MatoPlus) [inline pages/api/class/registrant/[classId].ts]: switch statement

File: utils/hooks/useStudentRegTableData.tsx [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+                accessor: "cityProvince",
+            },
+        ],
+        [],
+    );
+    const studentData = React.useMemo(
+        () =>
+            studentRegs?.map((reg) => {
+                return {
+                    // This should be a link
```
Comment (MatoPlus) [inline utils/hooks/useStudentRegTableData.tsx]: rm

**Pull request merged**